src/main/java/org/spongepowered/api/text/dictionary/Dictionary.java

```diff
@@ -0,0 +1,394 @@
+     * message could not be found
+     *
+     * @param translationId The translation id that could not be found
+     * @return A map containing the message not found text
+     */
+    protected Map<Locale, Text> getMissingTranslation(String translationId) {
+        final Map<Locale, Text> result = new HashMap<>();
+        final Text missing = Texts.builder("Missing message for translation id: " + translationId).color(TextColors.RED).build();
+        result.put(CommandSource.DEFAULT_LOCALE, missing);
+        return result;
+    }
+
+    /**
+     * Helper method to get the map containing the translations for the given
+     * translation id. This method will try to resolve the given translation id
+     * and if that is not possible it will try to use the message fallback
+     * function. If that fails too {@link #getMissingTranslation(String)} will
+     * be called.
+     *
+     * @param translationId The translation id to search
+     * @return A map containing the translations for the given translation id
+     */
+    protected Map<Locale, Text> getTranslations(final String translationId) {
+        Map<Locale, Text> translations = this.messages.get(translationId);
+        String currentTranslationId = translationId;
+        while (translations == null || translations.isEmpty()) {
+            currentTranslationId = this.messageFallback.apply(currentTranslationId);
+            if (currentTranslationId == null) {
+                return getMissingTranslation(translationId);
+            }
+            translations = this.messages.get(translationId);
+        }
+        return translations;
+    }
+
+    /**
+     * Gets the best fitting translation for the given translation id and
+     * locale.
+     *
+     * @param translationId The translation id to look for
+     * @param locale The requested language
+     * @return The best fitting translation for the given translation id and
+     *         locale
+     */
+    public Text getTranslation(String translationId, Locale locale) {
+        final Map<Locale, Text> translations = getTranslations(translationId);
+        final Text result = translations.get(locale);
+        if (result == null) {
+            return this.translationFallback.apply(locale, translations);
+        } else {
+            return result;
+        }
+    }
+
+    /**
+     * Gets a function that will return the best translation for the a given
+     * locale for the given translation id.
+     *
+     * @param translationId The translation id the function should return the
+     *        translations for
+     * @return A function that will return the best translation for the a given
+     *         locale for the given translation id
+     */
+    @Override
+    public Function<Locale, Text> apply(String translationId) {
+        return locale -> {
+            final Map<Locale, Text> translations = getTranslations(translationId);
```

Comment:
I refactored that method lately, until then the two methods weren't alike. And I haven't detected that yet.
Thanks for bringing this up to my attention.